File: backend-app/src/main/java/com/codeaddiction/entity/Candidate.java
package com.codeaddiction.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "candidate")
public class Candidate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String email;
    @Temporal(TemporalType.DATE)
    private Date dob;
    private  String partyName;
    private String address;
}

File: frontend-app/src/app/models/candidate.ts
export class Candidate {
  id?:number
  name:string
  email:string
  dob:string
  partyName:string
  address:string

  constructor(id: number, name: string, email: string, dob: string, partyName: string, address: string) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.dob = dob;
    this.partyName = partyName;
    this.address = address;
  }
}
